server/controllers/BookController.js

```diff
@@ -75,6 +75,65 @@ class BookController {
       return responseMessage(response, 500, { message: error.message });
     }
   }
+
+  /**
+* Edit Books
+* @name editBook
+* @param {Object} request
+* @param {Object} response
+* @returns {JSON} object
+* @memberof BookController
+*/
+  static async editBook(request, response) {
+    const { id } = request.params;
+    const {
+      title, description, isbn, price, yearPublished, authorId
+    } = request.body;
+    try {
+      const books = await Book.findAll();
+      const availableBook = books.find(book => book.id === +id);
+      if (!availableBook) return responseMessage(response, 400, { message: 'book does not exist' });
+      if (isbn) {
+        const availableIsbn = books.map(book => book.isbn).find(data => data === isbn);
+        if (availableIsbn === isbn) return responseMessage(response, 409, { message: 'isbn already exist' });
+      }
+      const results = await Book.update({
+        title, description, isbn, price, yearPublished
+      }, { where: { id: +id }, returning: true, plain: true });
```

Comment:
The plus (+) sign in front of id is not really necessary though. It would would without it,

Reply:
is necessary. without it the id is a string